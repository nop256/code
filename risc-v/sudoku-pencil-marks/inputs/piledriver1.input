6...4...5.3...1.4..2..3..1..9.....3.5..2....636.15.......9....78...2.......6.....
