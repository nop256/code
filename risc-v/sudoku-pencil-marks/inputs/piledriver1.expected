Please enter an 81-character board:

Testing pencil_marks with the following board:

+-----------------------+-----------------------+-----------------------+
|       | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 |       |
|   6     4 5 6   4 5 6 | 4 5 6     4     4 5 6 | 4 5 6   4 5 6     5   |
|       | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 |       | 1 2 3 |
| 4 5 6     3     4 5 6 | 4 5 6   4 5 6     1   | 4 5 6     4     4 5 6 |
| 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 |       | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 |       | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 |       | 1 2 3 |
| 4 5 6     2     4 5 6 | 4 5 6     3     4 5 6 | 4 5 6     1     4 5 6 |
| 7 8 9 |       | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 |       | 7 8 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 |
| 4 5 6     9     4 5 6 | 4 5 6   4 5 6   4 5 6 | 4 5 6     3     4 5 6 |
| 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       |
|   5     4 5 6   4 5 6 |   2     4 5 6   4 5 6 | 4 5 6   4 5 6     6   |
|       | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | 1 2 3 |       |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |
|   3       6     4 5 6 |   1       5     4 5 6 | 4 5 6   4 5 6   4 5 6 |
|       |       | 7 8 9 |       |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       |
| 4 5 6   4 5 6   4 5 6 |   9     4 5 6   4 5 6 | 4 5 6   4 5 6     7   |
| 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |
|   8     4 5 6   4 5 6 | 4 5 6     2     4 5 6 | 4 5 6   4 5 6   4 5 6 |
|       | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |
| 4 5 6   4 5 6   4 5 6 |   6     4 5 6   4 5 6 | 4 5 6   4 5 6   4 5 6 |
| 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |
+-----------------------+-----------------------+-----------------------+

The return value is 1
After pencil_marks returned the board is now:

+-----------------------+-----------------------+-----------------------+
|       | 1 . . | 1 . . | . . . |       | . 2 . | . 2 3 | . 2 . |       |
|   6     . . .   . . . | . . .     4     . . . | . . .   . . .     5   |
|       | 7 8 . | 7 8 9 | 7 8 . |       | 7 8 9 | 7 8 9 | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . . |       | . . . | . . . | . . . |       | . 2 . |       | . 2 . |
| . . .     3     . 5 . | . 5 .   . . 6     1   | . . 6     4     . . . |
| 7 . 9 |       | 7 8 9 | 7 8 . | 7 8 9 |       | 7 8 9 |       | . 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . . |       | . . . | . . . |       | . . . | . . . |       | . . . |
| 4 . .     2     4 5 . | . 5 .     3     . 5 6 | . . 6     1     . . . |
| 7 . 9 |       | 7 8 9 | 7 8 . |       | 7 8 9 | 7 8 9 |       | . 8 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 . |       | 1 2 . | . . . | . . . | . . . | 1 2 . |       | 1 2 . |
| 4 . .     9     4 . . | 4 . .   . . 6   4 . 6 | 4 5 .     3     4 . . |
| 7 . . |       | 7 8 . | 7 8 . | 7 8 . | 7 8 . | 7 8 . |       | . 8 . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | 1 . . | 1 . . |       | . . . | . . 3 | 1 . . | . . . |       |
|   5     4 . .   4 . . |   2     . . .   4 . . | 4 . .   . . .     6   |
|       | 7 8 . | 7 8 . |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | . 2 . |       |       | . . . | . 2 . | . 2 . | . 2 . |
|   3       6     4 . . |   1       5     4 . . | 4 . .   . . .   4 . . |
|       |       | 7 8 . |       |       | 7 8 9 | 7 8 9 | 7 8 9 | . 8 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 . | 1 . . | 1 2 3 |       | 1 . . | . . 3 | 1 2 3 | . 2 . |       |
| 4 . .   4 5 .   4 5 6 |   9     . . .   4 5 . | 4 5 6   . 5 6     7   |
| . . . | . . . | . . . |       | . 8 . | . 8 . | . 8 . | . 8 . |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | 1 . . | 1 . 3 | . . 3 |       | . . 3 | 1 . 3 | . . . | 1 . 3 |
|   8     4 5 .   4 5 6 | 4 5 .     2     4 5 . | 4 5 6   . 5 6   4 . . |
|       | 7 . . | 7 . 9 | 7 . . |       | 7 . . | . . 9 | . . 9 | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | 1 . . | 1 2 3 |       | 1 . . | . . 3 | 1 2 3 | . 2 . | 1 2 3 |
| 4 . .   4 5 .   4 5 . |   6     . . .   4 5 . | 4 5 .   . 5 .   4 . . |
| 7 . 9 | 7 . . | 7 . 9 |       | 7 8 . | 7 8 . | . 8 9 | . 8 9 | . 8 9 |
+-----------------------+-----------------------+-----------------------+

The return value is 0
After pencil_marks returned the board is now:

+-----------------------+-----------------------+-----------------------+
|       | 1 . . | 1 . . | . . . |       | . 2 . | . 2 3 | . 2 . |       |
|   6     . . .   . . . | . . .     4     . . . | . . .   . . .     5   |
|       | 7 8 . | 7 8 9 | 7 8 . |       | 7 8 9 | 7 8 9 | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . . |       | . . . | . . . | . . . |       | . 2 . |       | . 2 . |
| . . .     3     . 5 . | . 5 .   . . 6     1   | . . 6     4     . . . |
| 7 . 9 |       | 7 8 9 | 7 8 . | 7 8 9 |       | 7 8 9 |       | . 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . . |       | . . . | . . . |       | . . . | . . . |       | . . . |
| 4 . .     2     4 5 . | . 5 .     3     . 5 6 | . . 6     1     . . . |
| 7 . 9 |       | 7 8 9 | 7 8 . |       | 7 8 9 | 7 8 9 |       | . 8 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 . |       | 1 2 . | . . . | . . . | . . . | 1 2 . |       | 1 2 . |
| 4 . .     9     4 . . | 4 . .   . . 6   4 . 6 | 4 5 .     3     4 . . |
| 7 . . |       | 7 8 . | 7 8 . | 7 8 . | 7 8 . | 7 8 . |       | . 8 . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | 1 . . | 1 . . |       | . . . | . . 3 | 1 . . | . . . |       |
|   5     4 . .   4 . . |   2     . . .   4 . . | 4 . .   . . .     6   |
|       | 7 8 . | 7 8 . |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | . 2 . |       |       | . . . | . 2 . | . 2 . | . 2 . |
|   3       6     4 . . |   1       5     4 . . | 4 . .   . . .   4 . . |
|       |       | 7 8 . |       |       | 7 8 9 | 7 8 9 | 7 8 9 | . 8 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 . | 1 . . | 1 2 3 |       | 1 . . | . . 3 | 1 2 3 | . 2 . |       |
| 4 . .   4 5 .   4 5 6 |   9     . . .   4 5 . | 4 5 6   . 5 6     7   |
| . . . | . . . | . . . |       | . 8 . | . 8 . | . 8 . | . 8 . |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | 1 . . | 1 . 3 | . . 3 |       | . . 3 | 1 . 3 | . . . | 1 . 3 |
|   8     4 5 .   4 5 6 | 4 5 .     2     4 5 . | 4 5 6   . 5 6   4 . . |
|       | 7 . . | 7 . 9 | 7 . . |       | 7 . . | . . 9 | . . 9 | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | 1 . . | 1 2 3 |       | 1 . . | . . 3 | 1 2 3 | . 2 . | 1 2 3 |
| 4 . .   4 5 .   4 5 . |   6     . . .   4 5 . | 4 5 .   . 5 .   4 . . |
| 7 . 9 | 7 . . | 7 . 9 |       | 7 8 . | 7 8 . | . 8 9 | . 8 9 | . 8 9 |
+-----------------------+-----------------------+-----------------------+
